crates/rnote-engine/src/engine/rendering.rs

```diff
@@ -251,17 +250,23 @@ impl Engine {
         snapshot.append_node(
             gsk::ColorNode::new(
                 &gdk::RGBA::from_compose_color(self.document.config.background.color),
-                //&gdk::RGBA::RED,
                 &graphene::Rect::from_p2d_aabb(doc_bounds),
             )
             .upcast(),
         );
+        snapshot.pop();
 
-        for r in self.background_rendernodes.iter() {
-            snapshot.append_node(r);
+        if let (Some(bounds), Some(render_node)) = (
+            self.origin_background_rendernode,
+            self.background_rendernode.clone(),
+        ) {
+            snapshot.push_repeat(
+                &graphene::Rect::from_p2d_aabb(doc_bounds),
+                Some(&graphene::Rect::from_p2d_aabb(bounds)),
```

Comment:
it seems to me that we can get by by passing `None` for the child bounds argument here, because then the entire background tile rendernode size is used